duckduckgo/iOS
# Enable username&password passed in URL

Task/Issue URL: https://app.asana.com/0/414235014887631/1204080079383830/f
Tech Design URL: https://app.asana.com/0/481882893211075/1204065038916791/f
Security Triage URL: https://app.asana.com/0/1199892415909552/1204067721102886/f
BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/245
CC:

**Description**:
- Adds support to pass basic auth username/password in URL


**Steps to test this PR**:
URLs for testing: 
- https://authenticationtest.com/HTTPAuth/ (user:pass)
- http://httpbin.org/basic-auth/foo/bar (foo:bar)

1. Validate login/password passed in URL is used for basic authentication and is saved per session
2. Validate login/password passed in URL is not displayed in the UI or saved in browsing history
3. Validate when clicking links containing username/password (user:pass@domain.com), the credentials aren‘t displayed in UI but are used during basic auth
4. Validate if invalid credentials are provided then auth dialog is displayed
